### PR TITLE
Get latest AMI per OS

### PR DESCRIPTION
**Pull request by lukeseawalker**
Added filter to retrieve the latest AMI per OS if more than one is found with the specified cookbook and node git reference

Signed-off-by: Luca Carrogu <carrogu@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: util/generate-ami-list.py [hunk 1]
```diff
@@ -131,12 +131,26 @@ def get_images_ec2(filters, owner, region_name):
...
         ec2 = boto3.client("ec2", region_name=region_name)
         images = ec2.describe_images(Owners=[owner], Filters=filters)
-        return images
+        return latest_image_per_os(images)
     except ClientError:
         print("Warning: non authorized in region '{0}', skipping".format(region_name))
         pass
 
 
+def latest_image_per_os(images):
```
Comment (enrico-usai) [inline util/generate-ami-list.py]: minor: we can use the underscore prefix for private functions

Comment (enrico-usai) [reply util/generate-ami-list.py]: Minor: What about "filter_latest_os_images" or "get_latest_images"?

File: util/generate-ami-list.py [hunk 2]
```diff
@@ -131,12 +131,26 @@ def get_images_ec2(filters, owner, region_name):
...
+    # filter for the latest image per OS
+    images_filtered = {"Images": []}
+    for key, value in distros.items():
+        ami_filtered_and_sorted = sorted(
+            filter(lambda ami: ("-" + value + "-") in ami["Name"], images["Images"]),
```
Comment (enrico-usai) [inline util/generate-ami-list.py]: Minor: If this code is only executed by python3 you can write: 
```python
f"-{value}-"
```
or in any case you can use
```python
"-{0}-".format(value)"
```

File: util/generate-ami-list.py [hunk 3]
```diff
@@ -131,12 +131,26 @@ def get_images_ec2(filters, owner, region_name):
...
+        return latest_image_per_os(images)
     except ClientError:
         print("Warning: non authorized in region '{0}', skipping".format(region_name))
         pass
 
 
+def latest_image_per_os(images):
+    # filter for the latest image per OS
+    images_filtered = {"Images": []}
+    for key, value in distros.items():
```
Comment (enrico-usai) [inline util/generate-ami-list.py]: I know it is not related to your changes but `distros` is a global variable. Could you please define it uppercase?

File: util/generate-ami-list.py (hunk 1 above)
Comment (lukeseawalker) [reply util/generate-ami-list.py]: OK

File: util/generate-ami-list.py (hunk 3 above)
Comment (lukeseawalker) [reply util/generate-ami-list.py]: ok

File: util/generate-ami-list.py (hunk 2 above)
Comment (lukeseawalker) [reply util/generate-ami-list.py]: thanks

**Pull request merged**